### PR TITLE
docs: added doc to DomainMatrix

### PR DESCRIPTION
**Pull request by hyadav2k**
@ks147, kindly have a look at this PR, this will probably resolve the `sphinx` issues.

NO ENTRY

### CODE REVIEW

File: doc/src/modules/polys/DomainMatrix.rst [hunk 1]
```diff
@@ -0,0 +1,30 @@
+.. _polys-domainmatrix:
```
Comment (hyadav2k) [inline doc/src/modules/polys/DomainMatrix.rst:1]: Kindly remove this and add the changes you did in this file, and the index file.
Thanks.

**Comment by ks147** [comment]
It's not passing some of the tests. I am guessing you'll update it soon to fix the issue

**Comment by hyadav2k** [comment]
@ks147 I am not able to push the correct changes with the correct files.
I am asking you to please add the changes done in the `domainmatrix.py`, in your branch by manually adding the differences and then push them in your branch. It will probably fix the sphinx error.
Thanks.

**Comment by ks147** [comment]
> @ks147 I am not able to push the correct changes with the correct files.
> I am asking you to please add the changes done in the `domainmatrix.py`, in your branch by manually adding the differences and then push them in your branch. It will probably fix the sphinx error.
> Thanks.

Yeah, I've added the changes manually

**Pull request closed without merging**